src/lib/viem.ts

```diff
@@ -0,0 +1,202 @@
+	{"inputs":[],"name":"transfersEnabled","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"view","type":"function"},
+	{"stateMutability":"payable","type":"receive"}
+] as const;
+
+// Token information
+export const SNT_TOKEN = {
+	address: '0xE452027cdEF746c7Cd3DB31CB700428b16cD8E51' as Address,
+	name: 'Status Test Token',
+	symbol: 'STT',
+	decimals: 18
+} as const;
+
+// Network information
+export const network = {
+	name: 'Sepolia Testnet',
+	chainId: sepolia.id,
+	currency: 'ETH'
+};
+
+// Create a client for reading from the blockchain
+export const publicClient = createPublicClient({
+	chain: sepolia,
+	transport: http(rpcUrl)
+});
+
+// Wallet connection state stores
+export const walletAddress = writable<Address | undefined>(undefined);
+export const walletClient = writable<WalletClient | undefined>(undefined);
+export const walletBalance = writable<bigint | undefined>(undefined);
+export const sntBalance = writable<bigint | undefined>(undefined);
+export const sntError = writable<string | undefined>(undefined);
+
+// Derived store for formatted balances
+export const formattedBalance = derived(walletBalance, ($balance) => {
+	if ($balance === undefined) return undefined;
+	return Number(formatEther($balance)).toFixed(4);
+});
+
+export const formattedSntBalance = derived(sntBalance, ($balance) => {
+	if ($balance === undefined) return undefined;
+	return Number(formatUnits($balance, SNT_TOKEN.decimals)).toFixed(4);
+});
+
+// Function to fetch ETH balance
+async function fetchBalance(address: Address) {
+	try {
+		console.log('Fetching ETH balance for address:', address);
+		const balance = await publicClient.getBalance({ address });
+		console.log('Received ETH balance:', balance.toString());
+		walletBalance.set(balance);
+	} catch (error) {
+		console.error('Failed to fetch ETH balance:', error);
+		walletBalance.set(undefined);
+	}
+}
+
+// Function to fetch STT balance
+async function fetchSntBalance(address: Address) {
+	try {
+		console.log('Fetching STT balance for address:', address);
+		const balance = await publicClient.readContract({
+			address: SNT_TOKEN.address,
+			abi: CONTRACT_ABI,
+			functionName: 'balanceOf',
+			args: [address]
+		});
+		console.log('Received STT balance:', balance.toString());
+		sntBalance.set(balance);
+		sntError.set(undefined);
+	} catch (error) {
+		console.error('Failed to fetch STT balance:', error);
+		sntBalance.set(undefined);
+		if (error instanceof Error) {
+			sntError.set(error.message);
+		} else {
+			sntError.set('Failed to fetch STT balance');
+		}
+	}
+}
+
+// Function to connect wallet
+export async function connectWallet() {
+	if (!window.ethereum) {
+		throw new Error('MetaMask not installed');
```

Comment:
Same here. Again, just a nitpick, but you can choose from different wallets depending on what is installed in your browser